ssh/ssh_tunnel.py

```diff
@@ -66,11 +66,15 @@ def remote_cmd(self, cmd, timeout=None):
         run_cmd = self.ssh_cmd + ['-p', str(self.port), self.target] + cmd
         logger.debug('Running socket cmd: ' + ' '.join(run_cmd))
         try:
-            return check_output(run_cmd, timeout=timeout).decode('utf-8').rstrip('\r\n')
+            output = check_output(run_cmd, timeout=timeout)
+            if raw_output:
+                return output
+            else:
+                return output.decode('utf-8').rstrip('\r\n')
         except TimeoutExpired as e:
             logging.error('{} timed out after {} seconds'.format(cmd, timeout))
```

Comment:
Should use https://docs.python.org/3/library/logging.html#logging.Logger.exception

Also note, should just do a `raise`, that way you re-throw with all the attached stack history the most easily (otherwise you need to do a `raise ex from ex` thing).